distgen.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -e
+
+# before using this script, make sure your image repository contains:
+#   ./templates directory
+#   ./specs/multispec.yml file
+#   ./dg_manifest file
+
+# All templates should be stored in templates directory in container repository
+if [ ! -d "./templates" ]; then
+    echo "Templates for distgen not found"
+    exit
+fi
+
+# expecting file dg_manifest that contains variables:
+#   FILES_TO_COPY in form "path1:mode1 path2:mode2..."
+#   FILES_TO_GEN in form similar to FILES_TO_COPY
+#   INTERNAL_SYMLINKS in form "name1->target1 name1->target1..."
+if [ -f "./dg_manifest" ]; then
+    source ./dg_manifest
+fi
+
+WORKDIR=${PWD}
+DG=${DG:-/usr/bin/dg}
+DG_CONF=${DG_CONF:-centos-7-x86_64.yaml}
+
+cd templates
+OUTPUT_DIR=$(mktemp -d ${WORKDIR}/${OS}-${VERSION}-XXXX)
+echo "Copying files to ${OUTPUT_DIR}:"
+for file in ${FILES_TO_COPY}; do
+    split=(${file//:/ })
+    path=${split[0]}
+    mode=${split[1]}
+
+    echo " ${path} with mode - ${mode:-default}"
+    cp --parents ${path} ${OUTPUT_DIR}
+    if [ ! -z "${mode}" ]; then
+        chmod ${mode} ${OUTPUT_DIR}/${path}
+    fi
+done
+
+echo "Generating files:"
+for file in ${FILES_TO_GEN}; do
+    split=(${file//:/ })
+    path=${split[0]}
+    mode=${split[1]}
+
+    if [ ! -d $(dirname ${OUTPUT_DIR}/$path) ]; then
+        mkdir -p dirname ${OUTPUT_DIR}/$path
+    fi
+
+    # With max-passes > 1 is this directive non-effective
+    if grep "{%- raw %}" $path > /dev/null; then
+        max_passes=1
```

Comment:
Can you elaborate on this :-), I don't understand much what you try to achieve here....  The grep eats another fork, and is incomplete (one could use `{% raw %}` or `{%- raw -%}`).

Reply:
Problem description:
- some template files could contain some chars that are in fact jinja directives - e.g bash variable length vs. jinja comment - `${#myvar}`
- the directive {% raw %}/{% endraw %} fixes this problem, but the file can't be processed more than once
- while using `make distgen` we don't know whether template needs to be processed multiple times or just once.

Possible solutions
1,Add max-passes to manifest in form
```bash
# path:max-passes:mode
./Dockerfile:2:644
./script.sh::755
```
- with 1 as default - this could add complexity to UX

2, Expect jinja2.exceptions.TemplateSyntaxError - could be problem when there is an actual error

3, ... (please add your suggestion)

@praiskup @bkabrda what do you think?

Reply:
I actually already changed the mechanism of rerendering to not rerender the whole template, but just the spec values [1], which means this shouldn't be an issue going forward. I can do the release today and that would effectively make this problem go away, IIUC. @dhodovsk WDYT?

[1] https://github.com/devexp-db/distgen/pull/63

Reply:
@bkabrda I think it is great, would solve this issue nicely and also speed up the generation.

Reply:
I released distgen 0.20 which should fix this.